### ISSUE TITLE
1.20.2 unsupported

### ISSUE THREAD

**Issue by SlimeDog** [FancyMcPlugins/FancyHolograms#40]
Spigot 1.20.2-3788
FancyHolograms 2.0.1

On server start-up
```
[11:38:51] [Server thread/ERROR]: Could not load 'plugins/FancyHolograms-2.0.1.jar' in folder 'plugins'
org.bukkit.plugin.InvalidPluginException: java.lang.NoClassDefFoundError: io/papermc/paper/plugin/configuration/PluginMeta
        at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:149) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.loadPlugin(SimplePluginManager.java:394) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.loadPlugins(SimplePluginManager.java:301) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_20_R2.CraftServer.loadPlugins(CraftServer.java:437) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3882-Spigot-17ca32d-452fcb5]
        at net.minecraft.server.dedicated.DedicatedServer.e(DedicatedServer.java:219) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3882-Spigot-17ca32d-452fcb5]
        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:954) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3882-Spigot-17ca32d-452fcb5]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:298) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3882-Spigot-17ca32d-452fcb5]
        at java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NoClassDefFoundError: io/papermc/paper/plugin/configuration/PluginMeta
        at de.oliver.fancylib.serverSoftware.ServerSoftware.isFolia(ServerSoftware.java:15) ~[?:?]
        at de.oliver.fancyholograms.FancyHolograms.<init>(FancyHolograms.java:40) ~[?:?]
        at jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[?:?]
        at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[?:?]
        at java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128) ~[?:?]
        at jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:304) ~[?:?]
        at java.lang.Class.newInstance(Class.java:725) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:79) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:145) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        ... 7 more
Caused by: java.lang.ClassNotFoundException: io.papermc.paper.plugin.configuration.PluginMeta
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:147) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:99) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
        at de.oliver.fancylib.serverSoftware.ServerSoftware.isFolia(ServerSoftware.java:15) ~[?:?]
        at de.oliver.fancyholograms.FancyHolograms.<init>(FancyHolograms.java:40) ~[?:?]
        at jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62) ~[?:?]
        at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502) ~[?:?]
        at java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128) ~[?:?]
        at jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:304) ~[?:?]
        at java.lang.Class.newInstance(Class.java:725) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.<init>(PluginClassLoader.java:79) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.loadPlugin(JavaPluginLoader.java:145) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        ... 7 more
```

**Comment by OliverSchlueter** [comment]
Indeed Spigot and 1.20.2 are not supported yet.

**Comment by OliverSchlueter** [comment]
Try to use Paper when it's out

**Comment by SlimeDog** [comment]
> Try to use Paper when it's out

I will, when it's out. In the meantime, I'm testing plugins on Spigot.

**Comment by OliverSchlueter** [comment]
It's not working with Paper too.

**Comment by OliverSchlueter** [comment]
Plugin should work fine with paper 1.20.2 now

**Comment by SlimeDog** [comment]
I see changes to the API a week ago to support 1.20.2, but no release since August 21.

**Comment by SlimeDog** [comment]
And on my Mac, attempted build from current sources failed
```
./gradlew build
Downloading https://services.gradle.org/distributions/gradle-8.1-bin.zip
Welcome to Gradle 8.1!

Here are the highlights of this release:
 - Stable configuration cache
 - Experimental Kotlin DSL assignment syntax
 - Building with Java 20

For more details see https://docs.gradle.org/8.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':classes'.
> Could not create task ':processResources'.
   > Failed to retrieve the commit hash.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/8.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 57s
```

**Comment by OliverSchlueter** [comment]
I'm not very familiar with gradle too, but if you need the latest version, I usually run a jenkins job after I push (https://jenkins.fancyplugins.de/job/FancyHolograms/)

**Comment by SlimeDog** [comment]
Fixed in FH 2.0.2. Thanks.
